providers/apollo/metadata.go

```diff
@@ -0,0 +1,90 @@
+package apollo
+
+import (
+	"context"
+	"errors"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+// ListObjectMetadata creates metadata of object via reading objects using Apollo API.
+func (c *Connector) ListObjectMetadata(ctx context.Context,
+	objectNames []string,
+) (*common.ListObjectMetadataResult, error) {
+	// Ensure that objectNames is not empty
+	if len(objectNames) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	metadataResult := common.ListObjectMetadataResult{
+		Result: make(map[string]common.ObjectMetadata),
+		Errors: make(map[string]error),
+	}
+
+	for _, obj := range objectNames {
+		url, err := c.getAPIURL(obj)
+		if err != nil {
+			return nil, err
+		}
+
+		resp, err := c.Client.Get(ctx, url.String())
```

Comment:
Is there a way to get just one record? That way, we don't end up fetching hundreds of thousands of records that we do not need

Reply:
No, They are ids are random, 24 characters.

Reply:
For some objects, we can ensure that we are only returning a single page of records.

Reply:
@RajatPawar This is a possibility. Using pagination details(single page and limiting the records in a page), we can only return a single record. This would solve not returning so much data for the interest of one.